Adding a manpage for testasciidoc.py

aerostitch opened the pull request:
Hi,

In Debian, every distributed binaries are supposed to have a man page.
So as I've written this one for testasciidoc in Debian. Are you interested in integrating it to official asciidoc?
If so, can you validate this manpage looks good to you please?

Thanks in advance,
Joseph

apjanke commented on doc/testasciidoc.1.txt:
```diff
@@ -0,0 +1,120 @@
+
+COPYING
+-------
+Copyright \(C) 2014 Joseph Herlant. Free use of this software is
```
I think the copyright notice refers to the software being described, not the man page itself. Maybe this should have the same contents as the main `asciidoc` COPYING section?

apjanke commented on doc/testasciidoc.1.txt:
```diff
@@ -0,0 +1,120 @@
+
+COMMANDS
+--------
+The testasciidoc toolset has three different commands:
```
I think these are technically "subcommands", since `testasciidoc` is itself a command. Maybe "The testasciidoc command has three subcommands:"?

aerostitch commented:
Hi @apjanke,

Yes, that's what I set it to first, but @elextr told me to put my name instead.
I can even put something like "Asciidoc developpers" if you want, got no problem with it.

Joseph

elextr commented:
Hmmmm, the best reference I can find for manpage contents is http://man7.org/linux/man-pages/man7/man-pages.7.html which matches `man man-pages` on my system.  Copyright is not one of its headings, the only mention is a strong recommendation to put the copyright notice and authors info in the page source, not the output.  No mention of copyright of the software is made, so I suggest its not needed, it is after all officially in the source and should not be repeated here in case of mismatch.

aerostitch commented:
I couldn't find anything precise either about this. I think I'll just remove the section.

aerostitch commented:
up?

elextr commented on doc/testasciidoc.1.txt:
```diff
@@ -0,0 +1,114 @@
+
+RESOURCES
+---------
+SourceForge: <http://sourceforge.net/projects/asciidoc/>
```
This is no longer in use, it shouldn't be here, and feel free to provide PRs to remove it from anywhere else it appears :smile:

elextr commented:
Looks ok except the comment, but only by inspection, I can't test it ATM.

aerostitch commented:
Updated the branch and merged the 2 commits together so that it's cleaner.
Replaced the reference to sorceforge by the github one.
For the pull request you were talking about in your comment, where the package is downloadable from now if not sourceforge? (Note: the website still reference sourceforge and the googlecode website with mercurial, we might want to update that - http://asciidoc.org/INSTALL.html)

elextr commented:
>  (Note: the website still reference sourceforge and the googlecode website with mercurial, we might want to update that - http://asciidoc.org/INSTALL.html)

Its fixed in Git, but it needs to be pushed to the website.  I don't remember what the status of asciidoc.org was, @mojavelinux do you remember?

mojavelinux commented:
asciidoc.org is resolving to the gh-pages branch in this repository. So all you need to do to update the website is push to that branch.

Ideally, we should get Travis CI to do this step whenever a change is made to the repository (or, if we want to link it to a release, then whenever a release is cut).

There are lots of tutorials for how to get Travis CI to publish to GitHub Pages (just google for `travis ci publish to github pages`).

The key is to get Travis to be able to run the website build script. If you get that working, the rest is just following instructions in the tutorials. I'd be glad to help out, but I can't drive it.

mojavelinux commented:
At this point, we really need to see if Stuart will be willing to redirect http://www.methods.co.nz/asciidoc/ to asciidoc.org. There's just no good way as a community that we can manage the website on his server. GitHub Pages is ideally suited for that and we are 50% of the way there.

mojavelinux commented:
@elextr what I recommend is to first to a manual deployment to the gh-pages branch. Then we can get it automated. That fixes the immediate problem while we solve the long-term one.

What makes this extra difficult is that the asciidoc.org website doesn't use any known static site generator, of which there are like a million. It really is time that we start thinking about redoing the AsciiDoc website to give the format the recognition it deserves. The current website looks like it was created in the 80s and I hear that complaint often.

mojavelinux commented:
> @elextr what I recommend is to first to a manual deployment to the gh-pages branch. 

If we don't have instructions for that, then step 1 is finding and documenting the instructions for that.

mojavelinux commented:
> It really is time that we start thinking about redoing the AsciiDoc website to give the format the recognition it deserves.

Whether it's because of AsciiDoc Python or Asciidoctor, the AsciiDoc format is seeing greater adoption than ever before.

elextr commented:
> > It really is time that we start thinking about redoing the AsciiDoc website to give the format the recognition it deserves.

> Whether it's because of AsciiDoc Python or Asciidoctor, the AsciiDoc format is seeing greater adoption than ever before.

Yes, IMHO there needs to be a separation of *Asciidoc* and the (currently) two implementations Asciidoc Python and Asciidoctor making it clear that Asciidoc Python is a legacy implementation which is no longer under active development (but isn't going away soon and will accept fixes) and that Asciidoctor is a new implementation which is where new development is currently happening.

If Asciidoc Python is separated from Asciidoc the markup( which can have a nice new website), then I don't see much point is spending a lot of effort on the Asciidoc Python website, just fix the things that have changed.

All it needs is "somebody(s)" to do it.  :)

mojavelinux commented:
Well said. My thoughts exactly.
